src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+                                                             select repo).ToList();
+            if (!list.Any())
+            {
+                return previousTransactionDateTime;
+            }
+
+            Logger.LogInformation($"Going to create the github issue");
+            try
+            {
+                await CreateGitHubIssueAsync(previousTransactionDateTime.UtcDateTime,  list[0].Repository, list[0].Branch, 
+                    list[0].Action, list[0].ErrorMessage, list[0].Method , whereToCreateIssue);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Could not create a github issue for the error: " + list[0].ErrorMessage);
+            }
+            previousTransactionDateTime = list[0].Timestamp;
+            return previousTransactionDateTime;
+        }
+
+        public async Task<bool> SearchCreatedIssueAsync(DateTime transactionDateTime , string errorMessage)
+        {
+            var query = new StringBuilder();
+            HttpClient client = new HttpClient
+            {
+                BaseAddress = new Uri("https://api.github.com")
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", @"Mozilla/5.0 (Windows NT 10; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0");
+
+            query.Append($"is:issue+is:open+{"Error Message: " + errorMessage}in:title&order=desc");
+            JObject responseContent;
+            using (HttpResponseMessage response = await client.GetAsync(
+                $"search/issues?q={query}"))
+            {
+                responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+            }
+            JArray items = JArray.Parse(responseContent["items"].ToString());
+            if (!items.IsNullOrEmpty())
+            {
+                foreach (JObject item in items)
+                {
+                    var createdDate = item.GetValue("created_at");
+                    var result = createdDate.ToObject<DateTime>();
+                    if (result <= transactionDateTime)
+                    {
+                        Logger.LogInformation("The issue already exists so skipping this error");
+                        return false;
+                    }
+                    else
+                        return true;
+                }
+            }
+            return true;
+        }
+
+
+        private async Task CreateGitHubIssueAsync(DateTime transactionDateTime , string repositoryUrl, string branch, 
+            string action, String errorMessage, String method, string whereToCreateIssue)
+        {
+            Logger.LogInformation($"Something failed in the repository : {repositoryUrl} ");
+
+            string fyiHandles = "@epananth";
+            string gitHubToken = null, azureDevOpsToken = null;
+            string label = "UpdateDependency";
+
+            using (Logger.BeginScope($"Opening GitHub issue for RepositoryBranchUpdateHistory."))
```

Comment:
I think the "scope" should have some sort of identifier associated with it.  @alexperovich would know more.  Usually when we use it, the string is something like the message id... so that you can then search for all things related to this message id.  If it's always the same string, everything ends up in the same "scope"